Source/msg.cpp

```diff
@@ -614,81 +614,83 @@ void NetSendCmdExtra(TCmdGItem *p)
 	NetSendHiPri(MyPlayerId, (byte *)&cmd, sizeof(cmd));
 }
 
-DWORD OnWalk(TCmd *pCmd, Player &player)
+DWORD OnWalk(const TCmd *pCmd, Player &player)
 {
-	auto *p = (TCmdLoc *)pCmd;
+	const auto &message = *reinterpret_cast<const TCmdLoc *>(pCmd);
+	const Point position { message.x, message.y };
 
-	if (gbBufferMsgs != 1 && currlevel == player.plrlevel) {
+	if (gbBufferMsgs != 1 && currlevel == player.plrlevel && InDungeonBounds(position)) {

```

Comment:
If a message is invalid, I'd argue in favor of disconnecting the the other player, if hosting, or leaving the game, otherwise. Ignoring it simply sweeps a potential bigger problem under the rug, be it a bug, a corrupted/altered network packet potentially desyncing everything, a malicious client, or anything in between.

Reply:
From a network perspective it's probably safe enough to draw the conclusion, but considering that the engine often plays fast and loose with the bounds I would be a bit nervous that making it that unforgiving could result in some false positive disconnects. Maybe we can look in to it further after the release of 1.3.0.